src/CRSMatrix.hpp

```diff
@@ -169,31 +174,35 @@ class CRSMatrix : protected CRSMatrixView< T, COL_TYPE, INDEX_TYPE >
    * @brief @return A reference to *this reinterpreted as a CRSMatrixView< T, COL_TYPE, INDEX_TYPE const >.
    */
   constexpr inline
-  CRSMatrixView< T, COL_TYPE, INDEX_TYPE const > const & toView() const LVARRAY_RESTRICT_THIS
-  { return reinterpret_cast< CRSMatrixView< T, COL_TYPE, INDEX_TYPE const > const & >( *this ); }
+  CRSMatrixView< T, COL_TYPE, INDEX_TYPE const, BUFFER_TYPE > const &
+  toView() const LVARRAY_RESTRICT_THIS
+  { return reinterpret_cast< CRSMatrixView< T, COL_TYPE, INDEX_TYPE const, BUFFER_TYPE > const & >( *this ); }
```

Comment:
We do need to get rid of the `reinterpret_cast` at some point. This would be replaced by the construction of a view, and the return of that view rather than a reference. Can this be done without downstream changes?

Reply:
Yes everything should compile since we currently have
```c++
CRSMatrixView< T > const & CRSMatrix< T >::toView() const
{ return reinterpret_cast<CRSMatrixView< T > const & >( *this ); }
...
CRSMatrix< T > matrix;
CRSMatrixView< T > const & view = matrix.toView();
```
and if we change `toView() to return by value it will still work
```c++
CRSMatrixView< T > CRSMatrix< T >::toView() const
{ return CRSMatrixView< T >( m_entries, m_values, m_sizes, m_offsets ); }
...
CRSMatrix< T > matrix;
CRSMatrixView< T > const & view = matrix.toView();
```
However I think best practices dictate that now that `toView` returns by value we they type of `view` should be `CRSMatrixView< T > const` and not `CRSMatrixView< T > const &`.

Furthermore there could be runtime issues if the user writes code that assumes the view and the owner are the same object. Currently this would pass
```c++
Array1d< int > array;
ArrayView1d< int > const & view = array.toView();
array.resize( 1 );
EXPECT_EQ( view.size(), 1 );
EXPECT_NE( view.data(), nullptr );
```
However once we switch to returning by value this will fail.